agent/main.go

```diff
@@ -88,8 +91,15 @@ func main() {
 	flag.StringVar(&metricsbindaddress, "metricsbindaddress", ":8080", "metricsbindaddress is the TCP address that the controller should bind to for serving prometheus metrics.It can be set to \"0\" to disable the metrics serving")
 	flag.StringVar(&downloadpath, "downloadpath", "/var/lib/byoh/bundles", "File System path to keep the downloads")
 	flag.BoolVar(&skipInstallation, "skip-installation", false, "If you want to skip installation of the kubernetes component binaries")
+	flag.BoolVar(&printVersion, "version", false, "Print the version of the agent")
 	klog.InitFlags(nil)
 	flag.Parse()
+
+	if printVersion {
```

Comment:
Can we add tests to validate this?
You could probably add a new Context in this file [host_agent_test.go](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/agent/host_agent_test.go)